src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -68,22 +80,49 @@ def handle_bitstream(
         The information from sipi which is needed to establish a link from the resource
     """
     try:
-        if preprocessing_done:
-            resource_bitstream = resource.get_bitstream_information(bitstream.value, permissions_lookup)
-        else:
-            resource_bitstream = _upload_bitstream(
-                resource=resource,
-                sipi_server=sipi_server,
-                imgdir=imgdir,
-                permissions_lookup=permissions_lookup,
-            )
-            msg = f"Uploaded file '{bitstream.value}'"
-            print(f"{datetime.now()}: {msg}")
-            logger.info(msg)
+        resource_bitstream = _upload_bitstream(
+            resource=resource,
+            sipi_server=sipi_server,
+            imgdir=imgdir,
+            permissions_lookup=permissions_lookup,
+        )
+        msg = f"Uploaded file '{bitstream.value}'"
+        print(f"{datetime.now()}: {msg}")
+        logger.info(msg)
         return resource_bitstream
     except BaseError as err:
         err_msg = err.orig_err_msg_from_api or err.message
         msg = f"Unable to upload file '{bitstream.value}' of resource '{resource.label}' ({resource.id})"
         print(f"{datetime.now()}: WARNING: {msg}: {err_msg}")
         logger.warning(msg, exc_info=True)
         return None
+
+
+def _upload_bitstream(
```

Comment:
This function contains 10 LOC, and its call needs 6 LOC. So I wonder if it is worth factoring this out into an own function?

Reply:
I did it to make the distinction between the action of the upload and the error handling with the messaging. If we did not communicate at all with the user I could be convinced to combine them. But with the prints and logs as it is now it is too messy.